### PR TITLE
PP-699: Calls to memmove() should cast third argument need as size_t, not int

### PR DESCRIPTION
**Pull request by sid2364**
#### Issue-ID

* **[PP-699](https://pbspro.atlassian.net/browse/PP-699)**

#### Problem description
* Parameters to memmove() had third argument as int

#### Solution description
* Changed to size_t

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by arungrover** [comment]
Hey @sid2364  there are couple of other places where this change might be needed.
Example - in tpp_transport.c (add_pkts), req_quejob.c, qmgr.c (get_request) etc.
I'd say if you are making a change then make it in all other places too. What do you think?

**Comment by sid2364** [comment]
Sure @arungrover, will make the change and update the branch.

File: src/cmds/qsub.c [hunk 1]
```diff
@@ -2775,7 +2775,7 @@ fwd_tgt_creds(krb5_context context, krb5_auth_context auth_context, krb5_princip
 	if (!rhost)
 		return ENOMEM;
 	free_rhost = 1;
-	memcpy(rhost, server->data[1].data, server->data[1].length);
+	memcpy(rhost, server->data[1].data, (size_t)server->data[1].length);
```
Comment (sid2364) [inline src/cmds/qsub.c]: @arungrover There are a lot of calls to memcpy which don't have size_t cast to the last argument. Should I go ahead and change all of them? There are a lot of occurrences of this (~150), so I just want to confirm once before changing them all...

Comment (arungrover) [inline src/cmds/qsub.c]: @sid2364 the ticket was only about memmove, so let us just keep it to that for now. However log another bug if the same change is needed in memcpy as well.

Comment (sid2364) [inline src/cmds/qsub.c]: Okay.. will revert this change for memcpy.

**Comment by sid2364** [comment]
Rebased branch and squashed the commits.

**Comment by sid2364** [comment]
@PBSPro/pbspro-maintainers Have rebased the branch again. Please take this forward.

**Comment by mike0042** [comment]
Please add to ticket:
- Reason tests are not required
- Test logs
Then updated the checklist for this PR.

**Comment by sid2364** [comment]
Hi @mike0042, which test logs should I put up for this issue? This is a very small change, and is mostly precautionary, which is why I thought there were no tests required.

**Comment by mike0042** [comment]
@sid2364 Please add logs for the PTL smoke tests at a minimum. No new tests are required, but you must complete the checklist items prior to merge. That means also adding an explanation of why no new tests are required to the JIRA ticket.

**Comment by sid2364** [comment]
@mike0042 Sure, that makes sense. I'll add those details to the JIRA ticket.

**Comment by arungrover** [comment]
@sid2364 Can you please add smoke tests result to the ticket as mentioned by @mike0042 .

**Comment by sid2364** [comment]
@mike0042 @arungrover Added the smoke test results to the ticket.

**Comment by arungrover** [comment]
Merged
